ofrak_core/ofrak/core/elf/model.py

```diff
@@ -512,10 +513,10 @@ def get_section_index(self) -> Optional[int]:
     async def get_name(self) -> str:
         elf = await self.resource.get_only_ancestor_as_view(Elf, ResourceFilter.with_tags(Elf))
         string_section = await elf.get_string_section()
-        string_section_data = await string_section.resource.get_data(Range(self.st_name, Range.MAX))
-        name_string_end = string_section_data.find(b"\x00")
-        raw_symbol_name = string_section_data[:name_string_end]
-        return raw_symbol_name.decode("ascii")
+        ((_, raw_symbol_name),) = await string_section.resource.search_data(
+            re.compile(b".[^\x00]+\x00"), start=self.st_name, max_matches=1
```

Comment:
Same comment as above. This should probably only be compiled once. And the `.` seems unnecessary.

```suggestion
            re.compile(b"[^\x00]+\x00"), start=self.st_name, max_matches=1
```